src/controller/python/chip/clusters/command.cpp

```diff
@@ -168,6 +196,55 @@ PyChipError pychip_CommandSender_SendCommand(void * appContext, DeviceProxy * de
         usleep(busyWaitMs * 1000);
     }
 
+exit:
+    return ToPyChipError(err);
+}
+
+PyChipError pychip_CommandSender_SendGroupCommand(void * appContext, chip::GroupId groupId,
+                                                  chip::Controller::DeviceCommissioner * devCtrl, uint16_t timedRequestTimeoutMs,
+                                                  chip::ClusterId clusterId, chip::CommandId commandId, const uint8_t * payload,
+                                                  size_t length, uint16_t interactionTimeoutMs, uint16_t busyWaitMs)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    chip::Messaging::ExchangeManager * exchangeManager = chip::app::InteractionModelEngine::GetInstance()->GetExchangeManager();
+    VerifyOrReturnError(exchangeManager != nullptr, ToPyChipError(CHIP_ERROR_INCORRECT_STATE));
+
+    std::unique_ptr<GroupCommandSenderCallback> callback = std::make_unique<GroupCommandSenderCallback>(appContext);
+    std::unique_ptr<CommandSender> sender                = std::make_unique<CommandSender>(callback.get(), exchangeManager,
+                                                                            /* is timed request */ timedRequestTimeoutMs != 0);
+
+    app::CommandPathParams cmdParams = { groupId, clusterId, commandId, (app::CommandPathFlags::kGroupIdValid) };
+
+    SuccessOrExit(err = sender->PrepareCommand(cmdParams, false));
+
+    {
+        auto writer = sender->GetCommandDataIBTLVWriter();
+        TLV::TLVReader reader;
+        VerifyOrExit(writer != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+        reader.Init(payload, length);
+        reader.Next();
+        SuccessOrExit(writer->CopyContainer(TLV::ContextTag(to_underlying(CommandDataIB::Tag::kFields)), reader));
+    }
+
+    SuccessOrExit(err = sender->FinishCommand(timedRequestTimeoutMs != 0 ? Optional<uint16_t>(timedRequestTimeoutMs)
+                                                                         : Optional<uint16_t>::Missing()));
+
+    {
+        auto fabricIndex = devCtrl->GetFabricIndex();
+
+        chip::Transport::OutgoingGroupSession session(groupId, fabricIndex);
+        SuccessOrExit(err = sender->SendGroupCommandRequest(chip::SessionHandle(session)));
+    }
+
+    sender.release();
+    callback.release();
```

Comment:
Just so we are on the same page: At this point `sender` and `callback` are holding dangling pointers, because `OnDone` has already been called synchronously under `SendGroupCommandRequest` and the objects are deleted.  Releasing them like this is fine, of course, but you could also just skip the `callback` bit entirely, rely on the unique_ptr destructor to destroy `sender`, and call the "we are done" callback yourself after a successful call to `SendGroupCommandRequest`.  Or not have a "we are done" callback at all, since in practice you are done at stack unwind....

Either way, I guess; what's here is not wrong, just more complicated than it needs to be.